Recommendations refactor

WithoutPants opened the pull request:
- Changes the front page configuration to use a new `ui` settings interface. The backend shouldn't care about these UI-specific options, so adding a UI-specific interface that wasn't strictly graphql type-based seemed like the way to go, and will make it easier to extend this system in the future.
- Extends the supported "widget" types to include "pre-made" filters. These are a replacement for the existing hard-coded ones that appear initially.
- Changed the initial behaviour so that if there is no existing front page config value, then it is pre-populated based on the original hard-coded rows. These can then be customised by the user.
- Removed the old front page config interfaces.

WithoutPants committed:
Add generic UI settings

WithoutPants committed:
Back end bug fixes

WithoutPants committed:
Support adding premade filters

WithoutPants committed:
Clean up unused interfaces

WithoutPants committed:
Support images

cj12312021 commented on ui/v2.5/src/components/Images/ImageRecommendationRow.tsx:
```diff
@@ -0,0 +1,52 @@
+
+  return (
+    <RecommendationRow
+      className="gallery-recommendations"
```
We should probably use a new `image-recommendations` class for images that can inherit all the rules `gallery-recommendations` uses. This would enable users to have the ability to define their own rules specifically for images if they wanted.

cj12312021 commented on ui/v2.5/src/components/Images/ImageRecommendationRow.tsx:
```diff
@@ -0,0 +1,52 @@
+      >
+        {result.loading
+          ? [...Array(props.filter.itemsPerPage)].map((i) => (
+              <div key={i} className="gallery-skeleton skeleton-card"></div>
```
Same comment here for `gallery-skeleton`.

cj12312021 reviewed (commented):
These changes look great. I left some minor comments on the new `ImageRecommendationRow.tsx` file.

I also observed that when the Add window opens up for the first time, the `Add` button is not selectable despite there being a pre-selected filter. I have to select another different filter and then traverse back to the pre-selected filter to add it.
![Screenshot 2022-05-31 105143](https://user-images.githubusercontent.com/72030708/171229852-45841d01-8af8-426a-8542-9c77d99423a7.jpg)

You will see that `scenes` is lowercased in that same image. The same is true for all the content types selectable in Premade Filters. I'm not sure if this was intentional, but I think the first letter should be capitalized, so it doesn't stand out as much. It would also be consistent with the Nav tabs we have. I'll leave this up to you to decide though.

WithoutPants committed:
Set initial premade value

WithoutPants committed:
Correct image recommendation row classes

WithoutPants committed:
Fix internationalisation of premade filter names

WithoutPants commented:
> I also observed that when the Add window opens up for the first time, the `Add` button is not selectable despite there being a pre-selected filter. I have to select another different filter and then traverse back to the pre-selected filter to add it.

Thanks, this is now fixed.

> You will see that `scenes` is lowercased in that same image. The same is true for all the content types selectable in Premade Filters. I'm not sure if this was intentional, but I think the first letter should be capitalized, so it doesn't stand out as much. It would also be consistent with the Nav tabs we have. I'll leave this up to you to decide though.

This was a symptom of the object type not being internationalised properly. This is now also fixed.

Teda1 commented:
Thanks for making the updates. I see that no updates were made to the `ui/v2.5/src/components/FrontPage/styles.scss` file for the new image-skeleton class to use the same rules as gallery-skeleton. A `.recommendations-container .image-card hr` entry would likely need to be added in the section below to ensure the images display properly. I could make those updates in my pull request after merging yours into it if you want?

```
.recommendations-container .studio-card hr,
.recommendations-container .movie-card hr,
.recommendations-container .gallery-card hr {
  margin-top: auto;
}
```

WithoutPants committed:
Add missing class

WithoutPants commented:
All sorted. Thanks.
